### PR TITLE
Chore: app 3.0 buildout prep

### PR DESCRIPTION
**Pull request by mcous**
This PR cleans up a some of the build pipeline and dev dependencies on the UI side, hooks up redux with a very minimal robot state, and adds components based on early wireframes (including our very first reused React component 🥇)

CI is currently broken on this branch due to Python lint failures so it'll be addressed in a future PR if everyone's cool with that. If someone isn't cool with that, please let me know and/or push to this branch.

### CODE REVIEW

**Comment by mcous** [comment]
Ping @astaff and/or @IanLondon your reviews would be appreciated

File: app/.babelrc [hunk 1]
```diff
@@ -1,34 +1,34 @@
 {
   "env": {
     "test": {
-      "plugins":
-        ["istanbul"]
+      "plugins": [
+        "istanbul"
```
Comment (mcous) [inline app/.babelrc]: This plugin should be removed but it will be addressed in a PR that is waiting on this one to be merged

File: app/.babelrc [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
     "production": {
-      "presets": ["react-optimize"],
-      "plugins": ["babel-plugin-dev-expression"]
+      "presets": [
+        "react-optimize"
+      ]
     },
     "development": {
       "plugins": [
-        "transform-class-properties",
```
Comment (mcous) [inline app/.babelrc]: This transform shouldn't have been removed but is added back as a global transform (i.e. not just in the development environment) by the PR mentioned above

**Comment by IanLondon** [comment]
Why is build failing? https://travis-ci.org/OpenTrons/opentrons/jobs/263263469#L1083

**Comment by mcous** [comment]
Test is now failing due to old js unit tests (removed and re-written in branch `app-3-0-robot-comms` and what appears to be improperly setup integration tests (integration tests run without running python API).

I'm fine merging this as is and revisiting, but I can also spend the time to get the CI passing here if that makes people more comfortable...

**Comment by IanLondon** [comment]
After linting fixed, jest tests are failing https://travis-ci.org/OpenTrons/opentrons/jobs/264798098#L1229

**Comment by mcous** [comment]
Merging after after verbal 👍. Thanks for the review @IanLondon

**Pull request merged**